### DISCUSSION TITLE
log file not appearing for a vite launched electron app

### DISCUSSION THREAD

**Discussion by open768** [megahertz/electron-log#349] [Q&A]
I cant seem to find where electron-log writes the log file when an electron application is lauched using vite. (im using vutron as my scaffold)
I looked in my %USERPROFILE%\AppData\Roaming\ as suggested by the documentation (which in my case is on a network drive) and its not there. What is the correct way to expose the output filename?

However when the app is run outside of vite (in vanilla node) the log appears in the right place.
To find out where the file is written to, I modified the getFile function in main\transports\file\index.js file and it spat out the location

when running from vite, this code doesnt seem to get executed.
any help appreciated

**Comment by open768** [comment]
I added this code to find out whats happening and the file transport doesnt seem to be present when initialised from vite:

      for ( var [transName, transFn] of log.transportEntries(log.transports)){
        this.info(`log Transport: ${transName}`)

which gives
     17:44:18.307 ÔÇ║ WARN: ***** unable to find log file transport
     17:44:18.307 ÔÇ║ INFO: log Transport: console
     17:44:18.307 ÔÇ║ INFO: log Transport: ipc

ie no filetransport there!

**Reply by megahertz** [reply]
File transport isn't available in a renderer process. As for vite, I'm going to add [such a test](https://github.com/megahertz/electron-log/issues/350) later.

**Reply by open768** [reply]
I'm trying to figure out how to get the filetransport back. This is from the main process,  not renderer..  guidance appreciated

**Reply by megahertz** [reply]
I'm not sure what's the reason. Try to import 'electron-log/main' instead of 'electron-log'

**Reply by open768** [reply]
solved using a bit of a hack ;-)

    'use strict'

    import log from "electron-log"
    import fileTransportFactory from "electron-log/src/main/transports/file/index.js"
    import { getPathVariables } from "electron-log/src/main/transports/file/variables.js"

    // check if file transport is present
    if (!log.transports.file) {
        log.warn("unable to find log file transport - adding filetransport")
        log.transports.file = fileTransportFactory(log)
    }
    const oVars = getPathVariables(process.platform)
    log.info(`file will be output to ${oVars.libraryDefaultDir}`)

**Reply by open768** [reply]
problem was that in the vutron/vuetify/vite scaffold somewhere it sets the the main process to be called renderer, even though its not! 😁

**Reply by megahertz** [reply]
'electron-log/main' should be helpful in that case. Anyway, will check that with new e2e test

**Reply by megahertz** [reply]
Just tested, vite can't detect the main/renderer entry point correctly, so 'electron-log/main', 'electron-log/renderer' should be used.
